jax/_src/lax/linalg.py

```diff
@@ -381,13 +408,28 @@ def eig_batching_rule(batched_args, batch_dims, *, compute_left_eigenvectors,
                      compute_right_eigenvectors=compute_right_eigenvectors),
           (0,) * (1 + compute_left_eigenvectors + compute_right_eigenvectors))
 
+def eig_jvp_rule(primals, tangents, *, compute_left_eigenvectors,
+                 compute_right_eigenvectors):
+  if compute_left_eigenvectors or compute_right_eigenvectors:
+    raise NotImplementedError(
+        'The derivatives of eigenvectors are not implemented, only '
+        'eigenvalues. See '
+        'https://github.com/google/jax/issues/2748 for discussion.')
+  # Formula for derivative of eigenvalues w.r.t. a is eqn 4.60 in
+  # https://arxiv.org/abs/1701.00392
+  a, = primals
+  da, = tangents
+  l, v = eig(a, compute_left_eigenvectors=False)
```

Comment:
Rather than doing another linear solve, why not compute the left-eigenvectors as part of the eigenvalue problem and use those directly? I suspect that could save a bit of work.

Reply:
This is a really good idea but surprisingly doesn't seemt to work, because of the gauge subtlety rearing its ugly head again. `inv(v).T` is a valid left eigenvector matrix but it seems that it's not the matrix that eig produces (I just learned this from [here](https://math.stackexchange.com/questions/2071964/left-right-eigenvectors)). The tests fail with this approach; I tried doing `left_eigenvectors.T @ right_eigenvectors` in a debugger and it looked pretty far from identity.

Reply:
Oh, how awkward! 0.8 is pretty far from 1, well outside what I would consider a reasonable tolerance.